### PR TITLE
Feature: GSSAPI (Kerberos) support

### LINKED ISSUE #1608: Feature: GSSAPI (Kerberos) support
Hi there!

Kafka supports the Kerberos authentication protocol. Currently, Faststream does not explicitly specify a security mechanism for GSSAPI. 

Now, in order for the broker to work with Kafka servers with Kerberos authentication, you have to do this:
```python
broker._connection_kwargs.update(
    {
        "security_protocol": "SASL_SSL",
        "sasl_mechanism": "GSSAPI",
        "ssl_context": create_ssl_context(),
    },
)
```

I suggest creating a SASLGSSAPI class in [faststream/security.py](https://github.com/airtai/faststream/blob/main/faststream/security.py):
```python
class SASLGSSAPI(BaseSecurity):
    """Security configuration for SASL/GSSAPI authentication.

    This class defines security configuration for SASL/GSSAPI authentication.
    """

    def __init__(
        self,
        ssl_context: Optional["SSLContext"] = None,
        use_ssl: Optional[bool] = None,
    ) -> None:
        super().__init__(
            ssl_context=ssl_context,
            use_ssl=use_ssl,
        )

    def get_requirement(self) -> List["AnyDict"]:
        """Get the security requirements for SASL/GSSAPI authentication."""
        return [{"gssapi": []}]

    def get_schema(self) -> Dict[str, Dict[str, str]]:
        """Get the security schema for SASL/GSSAPI authentication."""
        return {"gssapi": {}}
```

and parsing in [kafka/security.py](https://github.com/airtai/faststream/blob/main/faststream/kafka/security.py) (and the same parsing in [confluent/security.py](https://github.com/airtai/faststream/blob/main/faststream/confluent/security.py)):
```python
def _parse_sasl_gssapi(security: SASLGSSAPI) -> "AnyDict":
    return {
        "security_protocol": "SASL_SSL" if security.use_ssl else "SASL_PLAINTEXT",
        "ssl_context": security.ssl_context,
        "sasl_mechanism": "GSSAPI",
    }
```

This will allow you to specify the authentication method in this way without interfering with protected attributes:
```python
broker = KafkaBroker(
    bootstrap_servers=[
        "example.ru:9092",
    ],
    sasl_kerberos_service_name="example_service_name",
    security=SASLGSSAPI(ssl_context=create_ssl_context(), use_ssl=True),
)
```

Well, I will also have to correct some shortcomings in specifying the sasl mechanism, for example [these](https://github.com/airtai/faststream/blob/main/faststream/confluent/client.py#L368).
I'm ready to implement this feature.

### PR DESCRIPTION
**Pull request by roma-frolov**
# Description
Fixed #1608 

## Type of change

- [x] New feature (a non-breaking change that adds functionality)

## Checklist

- [x] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [x] I have conducted a self-review of my own code
- [x] I have made the necessary changes to the documentation
- [x] My changes do not generate any new warnings
- [x] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [x] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [x] I have ensured that static analysis tests are passing by running `scripts/static-analysis.sh`
- [x] I have included code examples to illustrate the modifications

### CODE REVIEW

**Comment by roma-frolov** [comment]
I would like to clarify one point: aiokafka with `sasl_mechanism` == `GSSAPI` requires the `gssapi` library ([here](https://github.com/aio-libs/aiokafka/blob/e2afc034b59626242b51b06b30a62c565ef491d9/aiokafka/conn.py#L158)). This library is optional in aiokafka ([here](https://github.com/aio-libs/aiokafka/blob/e2afc034b59626242b51b06b30a62c565ef491d9/pyproject.toml#L42)). 

What to do here? Add the gssapi library as optional? Are additional checks needed?

File: docs/docs_src/kafka/security/sasl_gssapi.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import ssl
```
Comment (Lancetnik) [inline docs/docs_src/kafka/security/sasl_gssapi.py:1]: Why you added this file, but didn't use it anywhere?

**Review by Lancetnik — CHANGES_REQUESTED** [review]
Hi! Thank you for the PR - it looks really good. But, if you added docs/docs_src file, can you please include it to the documentation the same way with other ones? Also, we are trying to cover all such examples by test, so can you please add test for it as well (`tests/docs/etc` directory)?

File: docs/docs_src/kafka/security/sasl_gssapi.py (hunk 1 above)
Comment (roma-frolov) [reply docs/docs_src/kafka/security/sasl_gssapi.py:1]: I included this file in the documentation, but I already wrote the test earlier, [here it is](https://github.com/airtai/faststream/pull/1633/files#diff-b047124b26c2b25687a87a0062acb16d8822abd5f8addb0144db3de8c2864631R24). 
Is this enough?

Comment (Lancetnik) [reply docs/docs_src/kafka/security/sasl_gssapi.py:1]: Can u a test for it too? https://github.com/airtai/faststream/blob/main/tests/docs/kafka/test_security.py

Comment (roma-frolov) [reply docs/docs_src/kafka/security/sasl_gssapi.py:1]: done

**Review by Lancetnik — APPROVED** [review]

**Review by Lancetnik — APPROVED** [review]

**Comment by kumaranvpl** [comment]
@Lancetnik This PR added Kerberos support only for aiokafka broker, not for confluent broker. 
I have created a separate issue for it - https://github.com/airtai/faststream/issues/1669. I will create a PR for it in few mins.

**Pull request merged**